### ISSUE TITLE
Update Cart » Quantities Disabled » Invalid argument supplied for foreach() » includes/cart/actions.php on line 162

### ISSUE THREAD

**Issue by remcotolsma** [easydigitaldownloads/easy-digital-downloads#7518]
### Expected behavior
When I add a download with quantities disabled to the cart and change country on the checkout page and press the "Update Cart" button i don't expect to get an error.

### Actual behavior
When I add a download with quantities disabled to the cart and change country on the checkout page and press the "Update Cart" button i get an error:

```
Warning: Invalid argument supplied for foreach() in public_html/wp-content/plugins/easy-digital-downloads/includes/cart/actions.php on line 162

Warning: Cannot modify header information - headers already sent by (output started at public_html/wp-content/plugins/easy-digital-downloads/includes/cart/actions.php:162)
```

### Steps to reproduce the behavior
1) Create an Easy Digital Downloads post with quantities disabled
2) Add the download to the cart / checkout
3) Change country in the checkout
4) Press the "Update Cart" button

### Information (if a specific version is affected):
PHP Version: 7.2.26

EDD Version (or branch): 2.9.20

WordPress Version: 5.3.2

Any other relevant information:

Follow the code in following files:

- https://github.com/easydigitaldownloads/easy-digital-downloads/blob/2.9.20/templates/checkout_cart.php#L51-L55
- https://github.com/easydigitaldownloads/easy-digital-downloads/blob/2.9.20/includes/actions.php#L15-L36
- https://github.com/easydigitaldownloads/easy-digital-downloads/blob/2.9.20/includes/cart/actions.php#L155-L169

The fix is easy: check for `'edd-cart-downloads'` key in the `$data` array:

```php
if ( ! array_key_exists( 'edd-cart-downloads', $data ) ) {
	return;
}
```

```php
/**
 * Process cart updates, primarily for quantities
 *
 * @since 1.7
 */
function edd_process_cart_update( $data ) {
	if ( ! array_key_exists( 'edd-cart-downloads', $data ) ) {
		return;
	}

	foreach( $data['edd-cart-downloads'] as $key => $cart_download_id ) {
		$options  = json_decode( stripslashes( $data['edd-cart-download-' . $key . '-options'] ), true );
		$quantity = absint( $data['edd-cart-download-' . $key . '-quantity'] );
		edd_set_cart_item_quantity( $cart_download_id, $quantity, $options );
	}

}
add_action( 'edd_update_cart', 'edd_process_cart_update' );
```

**Comment by cklosowski** [comment]
@remcotolsma are cart quantities enabled for the store as well, as a global option and you're just disabling it on this product specifically? Just want to make sure we cover all the bases as we go through the testing process.

**Comment by remcotolsma** [comment]
I have the global "Cart Item Quantities" setting checked:

<img width="969" alt="Schermafbeelding 2020-01-20 om 21 11 29" src="https://user-images.githubusercontent.com/869674/72754949-868bde00-3bc9-11ea-9ad3-ee81e315a301.png">

But i think it's an issue in both scenarios. The hidden `edd-cart-downloads[]` field(s) are only rendered when quantities or enabled and not disabled per download:

```php
<?php if( edd_item_quantities_enabled() && ! edd_download_quantities_disabled( $item['id'] ) ) : ?>
```

https://github.com/easydigitaldownloads/easy-digital-downloads/blob/2.9.20/templates/checkout_cart.php#L51-L55

**Comment by cklosowski** [comment]
@remcotolsma I don't doubt both cases might be an issue, I just wanted to be sure to know where to start with replication so that once we start working on it, we don't have to reach back out.

**Comment by cklosowski** [comment]
Closing as this was fixed in #8219 via 9d2f793
